src/Psy/ConfigPaths.php

```diff
@@ -180,4 +180,41 @@ private static function getRealFiles(array $dirNames, array $fileNames)
 
         return $files;
     }
+
+    /**
+     * Touch $file mkdir and if $dir is not exists.
+     *
+     * Generates E_USER_NOTICE error if $file or $dir is not writable.
+     *
+     * @param string $dir
+     * @param string $file
+     *
+     * @return string|false Full path to $file. Return false if file is not writable.
+     */
+    public static function touchFileWithMkdir($dir, $file)
+    {
+        if (!is_writable($dir)) {
+            trigger_error("Writing to {$dir} is not allowed.", E_USER_NOTICE);
```

Comment:
If `$path` exists and is writable, but `$dir` isn't writable, we probably shouldn't treat that as an error.